src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -117,8 +119,13 @@ public void setEntityCallbacks(EntityCallbacks entityCallbacks) {
 		Assert.notNull(entity, "entity must not be null");
 		Assert.notNull(index, "index must not be null");
 
-		index(getIndexQuery(entity), index);
-		return entity;
+		IndexQuery query = getIndexQuery(entity);
+		index(query, index);
+
+		// suppressing because it's either entity itself or something of a correct type returned by an entity callback
+		@SuppressWarnings("unchecked")
+		T castResult = (T) query.getObject();
```

Comment:
This must not be null, see my comment for the `maybeCallbackAfterSave(T entity)` method.

Reply:
`query.getObject()` can be null either if `entity` is null (which is impossible, there is an assertion in the beginning of the method), or if `entityCallbacks.callback()` returns null (which is also impossible).
So the `castResult` cannot be null.

Would you still prefer to assert that is not null, just in case?